dtdl-tools/src/dtdl-validator/Directory.Build.props

```diff
@@ -1,7 +1,7 @@
 <Project>
   <PropertyGroup>
-    <BaseOutputPath>../../../target/debug/dtdl-validator/bin</BaseOutputPath>
-    <BaseIntermediateOutputPath>../../../target/debug/dtdl-validator/obj</BaseIntermediateOutputPath>
-    <MSBuildProjectExtensionsPath>../../../target/debug/dtdl-validator/obj</MSBuildProjectExtensionsPath>
+    <!--BaseOutputPath and BaseIntermediateOutputPath are adjusted when the project is run by a Cargo build. -->
+    <BaseOutputPath Condition="$(OUT_DIR) != ''">$(OUT_DIR)/$(BaseOutputPath)</BaseOutputPath>
```

Comment:
CodeQL needs to be able to successfully run the dotnet build (outside of Cargo), so that it can check the C# code for security issues.  I added the conditional (Condition="$(OUT_DIR) != ''") to allow the dotnet build to work outsiide of Cargo.  I have tested it and it is fine.